Add comparatives `__gt__` and  `__ge__`

oselz opened the pull request:
Add comparatives `__gt__` and `__ge__` (and by default `lt` and `le` as the sorting is symmetric)

akhundMurad commented:
@oselz Hello. Thanks for your contribution!
Could you check the lock file and update it? I added support for Python 3.12 in the main branch.

oselz commented:
> @oselz Hello. Thanks for your contribution! Could you check the lock file and update it? I added support for Python 3.12 in the main branch.

Hi @akhundMurad, I've rebased this PR to your updated main.  Hopefully should test ok now.

akhundMurad reviewed (approved):
Thank you
